app/connectors/HeaderGenerator.scala

```diff
@@ -20,14 +20,20 @@ import config.AppConfig
 import play.api.http.HeaderNames._
 import uk.gov.hmrc.http.HeaderCarrier
 import utils.Logger.logger
+import utils.UUIDGenerator
 
 import javax.inject.{Inject, Singleton}
 
 @Singleton
-class HeaderGenerator @Inject()(appConfig: AppConfig) {
+class HeaderGenerator @Inject()(appConfig: AppConfig, idGenerator: UUIDGenerator) {
 
   def headersForPEGA()(implicit hc: HeaderCarrier): Seq[(String, String)] = {
-    val headers = Seq("Environment" -> appConfig.pegaEnvironment)
+    val headers = Seq(
+      "CorrelationId" -> idGenerator.generateUUID,
+      "Environment" -> appConfig.pegaEnvironment
+    )
+    val correlationId = headers.find(_._1 == "CorrelationId").map(_._2).getOrElse("")
+    logger.info(s"CorrelationId  $correlationId")
```

Comment:
In reference to: https://github.com/hmrc/penalties-appeals-frontend/pull/139/files/a4029ce0d2549e134dad854441880c7f1a5cc2a7#r712122000 
```suggestion
    val correlationId = idGenerator.generateUUID
    val headers = Seq(
      "CorrelationId" -> correlationId,
      "Environment" -> appConfig.pegaEnvironment
    )
    logger.info(s"CorrelationId  $correlationId")
```